plugins/modules/cloud/misc/proxmox_disk.py

```diff
@@ -0,0 +1,704 @@
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+DOCUMENTATION = r'''
+---
+module: proxmox_disk
+short_description: Management of a disk of a Qemu(KVM) VM in a Proxmox VE cluster.
+version_added: 5.4.0
+description:
+  - Allows you to perform some supported operations on a disk in Qemu(KVM) Virtual Machines in a Proxmox VE cluster.
+author: "Castor Sky (@castorsky) <csky57@gmail.com>"
+options:
+  name:
+    description:
+      - The (unique) name of the VM.
+      - Required only one of C(name) or C(vmid).
+    type: str
+  vmid:
+    description:
+      - The (unique) ID of the VM.
+    type: int
+  disk:
+    description:
+      - The disk key (C(unused[n]), C(ide[n]), C(sata[n]), C(scsi[n]) or C(virtio[n])) you want to operate on.
+    type: str
+    required: True
+  state:
+    description:
+      - Indicate desired state of the disk. C(absent) can be used only with C(remove) action
+    type: str
+    choices: ['present', 'absent']
+    default: present
+  action:
+    description:
+      - Action to perform on the disk.
+    type: str
+    choices: ['create', 'update', 'grow', 'detach', 'move', 'remove']
+    required: True
+  force_replace:
+    description:
+      - Force replace existing attached disk with the new one leaving old disk unused
+    type: bool
+    default: False
+  storage:
+    description:
+      - The drive's backing storage.
+      - Used primarily in C(create) and C(move) actions.
+    type: str
+  size:
+    description:
+      - For create action C(size) is desired volume size in GB to allocate.
+      - >
+        For grow action C(size) is the new size of volume. With the C(+) sign
+        the value is added to the actual size of the volume and without it, the value is taken as an absolute one.
+    type: str
+  bwlimit:
+    description:
+      - Override I/O bandwidth limit (in KB/s).
+      - Used only in C(move) action
+    type: int
+  delete_moved:
+    description:
+      - Delete the original disk after successful copy.
+      - By default the original disk is kept as unused disk.
+      - Used only in C(move) action
+    type: bool
+  target_disk:
+    description:
+      - The config key the disk will be moved to on the target VM (for example, C(ide0) or C(scsi1)).
+      - Default is the source disk key.
+      - Used only in C(move) action
+    type: str
+  target_storage:
+    description:
+      - Target storage.
+      - Used only in C(move) action
+    type: str
+  target_vmid:
+    description:
+      - The (unique) ID of the VM.
+      - Used only in C(move) action
+    type: int
+  timeout:
+    description:
+      - Timeout for move action
+      - Used only in C(move) action
+    type: int
+    default: 600
+  aio:
+    description:
+      - AIO type to use.
+    type: str
+    choices: ['native', 'threads', 'io_uring']
+  backup:
+    description:
+      - Whether the drive should be included when making backups.
+    type: bool
+  bps:
```

Comment:
If I may suggest, it might make more sense to group all the `bps*`, `iops*`, and `mbps*` parameters into three params `bps`, `iops`, `mbps`, of type `dict` and pass these different parameters as suboptions to those.

Reply:
By design module is bypassing basic options to API as is, so making sub-options with bandwidth limiters will need additional parsing.
If you think that it will simplify usage of module I will isolate proposed options into separate dict.

Reply:
It was more a matter of organizing the information in the input, but I see what you mean and indeed, it would add an unnecessary complexity to the module.